src/transformers/hf_argparser.py

```diff
@@ -126,6 +126,9 @@ def parse_args_into_dataclasses(
         if return_remaining_strings:
             return (*outputs, remaining_args)
         else:
+            if remaining_args:
+                raise ValueError(f"Some specified arguments are not used by the trainer: {remaining_args}")
```

Comment:
```suggestion
                raise ValueError(f"Some specified arguments are not used by the HfArgumentParser: {remaining_args}")
```

(because it's used in non-Trainer use cases)